h/views/activity.py

```diff
@@ -42,9 +42,81 @@ def search(request):
     # Fetch results
     result = query.execute(request, q, page_size=page_size)
 
+    for user in result.aggregations.get('users', []):
+        user['username'] = util.user.split_user(user['user'])['username']
+        del user['user']
+
     return {
         'total': result.total,
         'aggregations': result.aggregations,
         'timeframes': result.timeframes,
         'page': paginate(request, result.total, page_size=page_size),
     }
+
+
+@view_config(route_name='activity.group_search',
+             request_method='GET',
+             renderer='h:templates/activity/search.html.jinja2')
+def group_search(request):
+    if not request.feature('search_page'):
+        raise httpexceptions.HTTPNotFound()
+
+    result = search(request)
+
+    pubid = request.matchdict['pubid']
+
+    try:
+        group = request.db.query(models.Group).filter_by(pubid=pubid).one()
+    except exc.NoResultFound:
+        return result
```

Comment:
It would probably be better if the groups service had a method for getting a group by its pubid, rarther than the view function accessing sqlalchemy itself here, but it doesn't and I'm not 100% sure that adding it won't be controversial and I'm in a bit of a hurry so I'm not fixing that here.